ibis_substrait/compiler/translate.py

```diff
@@ -1081,48 +977,28 @@ def set_op(
     )
 
 
-@translate.register(ops.Aggregation)
-def aggregation(
-    op: ops.Aggregation,
+@translate.register(ops.Aggregate)
+def aggregate(
+    op: ops.Aggregate,
     *,
     compiler: SubstraitCompiler,
     **kwargs: Any,
 ) -> stalg.Rel:
-    if op.having:
-        raise NotImplementedError("`having` not yet implemented")
-
-    table = op.table.to_expr()  # type: ignore
-    predicates = [pred.to_expr() for pred in op.predicates]  # type: ignore
-    input = translate(
-        table.filter(predicates) if predicates else table,
-        compiler=compiler,
-        **kwargs,
-    )
-
-    # TODO: we handle sorts before aggregation here but we shouldn't be
-    #
-    # sorts should be compiled after after aggregation, because that is
-    # semantically where they belong, but due to ibis storing sort keys in the
-    # aggregate class we have to sort first, so ibis will use the correct base
-    # table when decompiling
-    if op.sort_keys:
-        sorts = [translate(key, compiler=compiler, **kwargs) for key in op.sort_keys]
-        input = stalg.Rel(sort=stalg.SortRel(input=input, sorts=sorts))
-
     aggregate = stalg.AggregateRel(
-        input=input,
+        input=translate(op.parent, compiler=compiler, **kwargs),
         groupings=[
             stalg.AggregateRel.Grouping(
                 grouping_expressions=[
-                    translate(by, compiler=compiler, **kwargs) for by in op.by
+                    translate(v, compiler=compiler, **kwargs)
+                    for k, v in op.groups.items()
                 ]
             )
         ],
         measures=[
             stalg.AggregateRel.Measure(
-                measure=translate(metric, compiler=compiler, **kwargs)
+                measure=translate(v, compiler=compiler, **kwargs)
             )
-            for metric in op.metrics
+            for k, v in op.metrics.items()
```

Comment:
```suggestion
            for agg_func in op.metrics.keys()
```

Reply:
This should be .values(), not .keys(). I'll fix in a follow-up

Reply:
ahh, yeah, my mistake there